pkg/subnet/subnet.go

```diff
@@ -94,10 +94,12 @@ func WriteSubnetFile(path string, config *Config, ipMasq bool, sn ip.IP4Net, ipv
 
 	fmt.Fprintf(f, "FLANNEL_MTU=%d\n", mtu)
 	_, err = fmt.Fprintf(f, "FLANNEL_IPMASQ=%v\n", ipMasq)
-	f.Close()
 	if err != nil {
 		return err
 	}
+	if err := f.Close(); err != nil {
```

Comment:
If you move `f.Close()` after the return, it is possible that we never reach f.Close() and we leave the file open. In this case, I think the best is to create a function that we can "defer" where the file is closed and we assess if there is an error while closing the file

Reply:
I think each of these `fmt.Fprintf(`s deserve a err handler, instead I'm thinking of using a buffer to write this in one shot.

I'll add a close() whose err is not handled to capture the leaks.